internal/controller/accessrequest_controller.go

```diff
@@ -768,41 +768,63 @@ func ProjectChangeShouldTriggerReconcile(newProj, oldProj *argocd.AppProject) bo
 		return true
 	}
 
-	for _, newRole := range newProj.Spec.Roles {
-		if !slices.ContainsFunc(oldProj.Spec.Roles, func(oldRole argocd.ProjectRole) bool {
-			if len(newRole.Policies) != len(oldRole.Policies) {
-				return false
-			}
-			if len(newRole.JWTTokens) != len(oldRole.JWTTokens) {
+	for _, oldRole := range oldProj.Spec.Roles {
+		if !HasRole(newProj.Spec.Roles, oldRole) {
+			return true
+		}
+	}
+	return false
+}
+
+// HasRole checks if the given role exists in the list of roles.
+// It compares the role's Name and Description, and ensures that the Policies,
+// JWTTokens, and Groups slices are of equal length and contain the same elements.
+// Returns true if an equivalent role is found, otherwise returns false.
+func HasRole(roles []argocd.ProjectRole, role argocd.ProjectRole) bool {
```

Comment:
switch HasRole and MatchRolePoliciesAndTokens to private/unexported since they are utility methods and don't have dedicated unit test